src/Collections/TerminusCollection.php

```diff
@@ -0,0 +1,275 @@
+        if (isset($options['fetch_args'])) {
+            $args = array_merge($args, $options['fetch_args']);
+        }
+
+        if ($this->paged) {
+            $results = $this->request()->pagedRequest($this->getUrl(), $args);
+        } else {
+            $results = $this->request()->request($this->getUrl(), $args);
+        }
+
+        return $results['data'];
+    }
+
+    /**
+     * Returns an array of data where the keys are the attribute $key and the
+     *   values are the attribute $value, filtered by the given array
+     *
+     * @param array $filters Attributes to match during filtration
+     *   e.g. array('category' => 'other')
+     * @param string $key Name of attribute to make array keys
+     * @param string|array $value Name(s) of attribute to make array values
+     * @return array Array rendered as requested
+     *         $this->attribute->$key = $this->attribute->$value
+     */
+    public function getFilteredMemberList(
+        array $filters,
+        $key = 'id',
+        $value = 'name'
+    ) {
+        $members = $this->getMembers();
+        $member_list = [];
+
+        $values = $value;
+        if (!is_array($values)) {
+            $values = [$value,];
+        }
+        foreach ($members as $member) {
+            $member_list[$member->get($key)] = [];
+            foreach ($values as $item) {
+                $member_list[$member->get($key)][$item] = $member->get($item);
+            }
+            if (count($member_list[$member->get($key)]) < 2) {
+                $member_list[$member->get($key)] =
+                    array_pop($member_list[$member->get($key)]);
+            }
+            foreach ($filters as $attribute => $match_value) {
+                if ($member->get($attribute) != $match_value) {
+                    unset($member_list[$member->get($key)]);
+                    break;
+                }
+            }
+        }
+        return $member_list;
+    }
+
+    /**
+     * Returns an array of data where the keys are the attribute $key and the
+     *   values are the attribute $value
+     *
+     * @param string $key Name of attribute to make array keys
+     * @param string $value Name of attribute to make array values
+     * @return array Array rendered as requested
+     *         $this->attribute->$key = $this->attribute->$value
+     */
+    public function getMemberList($key = 'id', $value = 'name')
+    {
+        $member_list = $this->getFilteredMemberList([], $key, $value);
+        return $member_list;
+    }
+
+    /**
+     * Retrieves all members of this collection
+     *
+     * @return TerminusModel[]
+     */
+    protected function getMembers()
+    {
+        if (empty($this->models)) {
+            $this->fetch();
+        }
+        return $this->models;
+    }
+
+    /**
+     * Get the class that this collection collects.
+     *
+     * @return string
+     */
+    protected function getCollectedClass()
```

Comment:
This getter function returns a property which has the same visibility as the function.